### ISSUE TITLE
[BUG] Warnings after import pyrolite.plot

### ISSUE THREAD

**Issue by tectoni** [morganjwilliams/pyrolite#71]
**Describe the bug**
Numerous warnings appear at 'import pyrolite.plot'

**To Reproduce**
Steps to reproduce the behavior: -> see output below

**Expected behavior**
import without warnings 

**Environment:**
 - OS: macOS 11.6.1
 - Python Version 3.9.12 (and the same with 3.8)
 - Pyrolite Version 0.3..1 installed with pip in the environment

**Additional Context**
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:23:20) 
[Clang 12.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib.pyplot as plt
>>> import pandas as pd
>>> import pyrolite.plot
2022-03-27 15:48:53,513 pyrolite.util.plot.helpers - WARNING: scikit-learn not installed
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Note 
----- in 
Plots spidergrams for trace elements data. Additional arguments are typically forwarded
to respective :mod:`matplotlib` functions :func:`~matplotlib.pyplot.plot` and... in the docstring of spider in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/spider.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Method for plotting histograms (mode='hist2d'|'hexbin') or kernel density
esitimates from point data. Convenience access function to... in the docstring of density in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Note 
----- in 
Creates diagramatic representation of data density and/or frequency for either
binary diagrams (X-Y) or ternary plots.... in the docstring of density in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/density/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Create a :func:`pyrolite.plot.parallel.parallel`. coordinate plot from
the columns of the :class:`~pandas.DataFrame`.... in the docstring of parallel in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Create a parallel coordinate plot across dataframe columns, with
individual lines for each row.... in the docstring of parallel in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/parallel.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Todo 
------ in 
Create a parallel coordinate plot across dataframe columns, with
individual lines for each row.... in the docstring of parallel in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/parallel.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
------------ in 
Pass the pandas object to :func:`pyrolite.plot.spider.REE_v_radii`.
... in the docstring of REE in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Note 
----- in 
Creates an axis for a REE diagram with ionic radii along the x axis.
... in the docstring of REE_v_radii in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/spider.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Convenience method for scatter plots using the pyroplot API. See
further parameters for `matplotlib.pyplot.scatter` function below.... in the docstring of scatter in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Convenience method for line plots using the pyroplot API. See
further parameters for `matplotlib.pyplot.scatter` function below.... in the docstring of plot in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Method for spider plots. Convenience access function to
:func:`~pyrolite.plot.spider.spider` (see `Other Parameters`, below), where... in the docstring of spider in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Todo 
----- in 
Method for spider plots. Convenience access function to
:func:`~pyrolite.plot.spider.spider` (see `Other Parameters`, below), where... in the docstring of spider in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Method for creating stem plots. Convenience access function to
:func:`~pyrolite.plot.stem.stem` (see `Other Parameters`, below), where... in the docstring of stem in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Parameters 
----------- in 
Create a stem (or 'lollipop') plot, with optional orientation.
... in the docstring of stem in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/stem.py.
  warn(msg)
/Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/numpydoc/docscrape.py:434: UserWarning: potentially wrong underline length... 
Note 
----- in 
Convenience method for scatter plots using the pyroplot API. See
further parameters for `matplotlib.pyplot.scatter` function below.... in the docstring of scatter in /Users/pappel/opt/anaconda3/envs/myenv/lib/python3.9/site-packages/pyrolite/plot/__init__.py.
  warn(msg)
>>>

**Comment by tectoni** [comment]
Could not reproduce issue on another machine, all ok there. Must be a problem of the local conda setup.

**Comment by morganjwilliams** [comment]
Hey @tectoni, thanks for the report - this is a bug which propagated after some updates in `numpy`/`numpydoc` (so will depend on respective verisons of that), partly due to how docstrings are modified in `pyrolite` - should be fixed on `develop` and out in the next release.

**Comment by seignovert** [comment]
Hi @tectoni and @morganjwilliams,

I'm facing the same issue on a brand new conda install (with python 3.9.13 on macOS).
Could you provide some guidance regarding the proper `numpy` and `numpydoc` version to use with this package to avoid all these warnings?

Thanks!

**Comment by morganjwilliams** [comment]
Hey @seignovert, thanks for the report; I'll see if I can replicate (although not on MacOS) and debug some more before I get back to you.

**Comment by morganjwilliams** [comment]
I don't currently see the issue on Windows (Python 3.9.13, `pyrolite` v0.3.1+dev, `numpydoc` v1.4.0), but will debug more on Linux which should be closer to MacOS (so far on Linux I don't see it with Python 3.7.12, `pyrolite` v0.3.1+dev, `numpydoc` v1.2.1 but need to check Python 3.9 and corresponding `numpydoc` there).

**Comment by seignovert** [comment]
@morganjwilliams Here are the details on my side.
- `python` 3.9.13
- `pyrolite` 0.3.1
- `numpy` 1.22.4
- `numpydoc` 1.4.0

I was able to replicate this issue on an other macOS system and on a Linux system too (Fedora 8.5):
```bash
conda create -n test-pyrolite python=3.9.13
conda activate test-pyrolite
pip install pyrolite==0.3.1
python -c 'import pyrolite.plot'
```
```
2022-06-16 10:43:20,341 pyrolite.util.plot.helpers - WARNING: scikit-learn not installed
/miniconda/envs/test-pyrolite/lib/python3.9/site-packages/numpydoc/docscrape.py:449: UserWarning: potentially wrong underline length... 
Note 
----- in 
Plots spidergrams for trace elements data. Additional arguments are typically forwarded
to respective :mod:`matplotlib` functions :func:`~matplotlib.pyplot.plot` and... in the docstring of spider in /miniconda/envs/test-pyrolite/lib/python3.9/site-packages/pyrolite/plot/spider.py.
  warn(msg)
/miniconda/envs/test-pyrolite/lib/python3.9/site-packages/numpydoc/docscrape.py:449: UserWarning: potentially wrong underline length
... 
```

**Comment by morganjwilliams** [comment]
Thanks @seignovert! Note that installing the current release as in your snippet would indeed be expected to have this behavior; using the `develop` branch via git should avoid it (`pip install git+https://github.com/morganjwilliams/pyrolite.git@develop#egg=pyrolite`) - but the fixes still haven't been released. Can you clarify whether your `pyrolite` version shows a full version (i.e. `0.3.1`), or shows the develop tag (`0.3.1+26.g100cb1c`, or similar)? If the latter and you're still seeing these pop up then I've missed something in the updates there.

**Comment by tectoni** [comment]
I was facing the same problems, but later could not reproduce the issue on another mac with a similar setup. I removed my Anaconda install completely an re-installed and the issue was gone.

**Comment by seignovert** [comment]
Ok. I recreated a new conda env and install the `develop` version with `pip` this time (I used `git` in my previous test) and now it works:
```bash
conda create -n test-pyrolite python=3.9.13
conda activate test-pyrolite
pip install git+https://github.com/morganjwilliams/pyrolite.git@develop#egg=pyrolite
pip show pyrolite # -> 0.3.1+26.g100cb1c
python -c 'import pyrolite.plot'
```
Only the `scikit-learn` warning is still present but that's not the focus of this issue.
```
2022-06-16 15:12:14,345 pyrolite.util.plot.helpers - WARNING: scikit-learn not installed
```

It might be nice to push a 0.3.2 patch with dev branch maybe?

Anyway, thanks for the quick support! 👏

**Comment by morganjwilliams** [comment]
Yeah, there are a few plot helpers which use `sklearn`, so that last warning is to be expected if you don't have it (shouldn't affect your use day-to-day for the most part). Apologies for the delay on 0.3.2, there were a few things lined up for the release but they didn't happen as planned. I'll see if I can wrap up some things for next week 🤞 and defer the planned features for now.

**Comment by morganjwilliams** [comment]
@tectoni @seignovert this one should be fixed now v0.3.2 is out, apologies it took a bit of time to get around to!
